src/App.tsx

```diff
@@ -1,24 +1,184 @@
 import React from 'react';
 import './App.scss';
 
-const goodsFromServer: string[] = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
-
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+type State = {
+  selectedGood: string;
+};
+
+export class App extends React.Component<{}, State> {
+  goodsFromServer = [
+    'Dumplings',
+    'Carrot',
+    'Eggs',
+    'Ice cream',
+    'Apple',
+    'Bread',
+    'Fish',
+    'Honey',
+    'Jam',
+    'Garlic',
+  ];
+
+  state = {
+    selectedGood: '',
+  };
+
+  componentDidMount() {
+  }
+
+  componentDidUpdate(prevState: State) {
+    if (prevState.selectedGood !== this.state.selectedGood) {
+      const initialText = document.querySelector('.initial_text');
+      const stateLength = this.state.selectedGood.split(',').length;
+
+      if (initialText !== null) {
+        if (stateLength === 1 && this.state.selectedGood !== '') {
+          initialText.textContent = `${this.state.selectedGood.replaceAll('_', ' ')} is selected`;
+        } else if (this.state.selectedGood === '') {
+          initialText.textContent = 'No goods selected';
+        } else {
+          initialText.textContent = `${this.state.selectedGood.replaceAll('_', ' ')} are selected`;
+        }
+      }
+    }
```

Comment:
It's really hard to read this part, simplify it, all you need to check - is to check selectedGoods.length using switch, with 3 cases:
- length = 0 showing `No goods are selected`
- length = 1 showing `{goods[0]} is selected`
- length > 1 showing `a, b, c and d are selected`